File: cml-compiler/cml-generator/src/test/resources/templates/lang/py/expressions/employee.py
class Employee:

    _employment = _Employment()

    def __init__(self, number: 'int', name: 'str', employer: 'Organization') -> 'None':
        self.__number = number
        self.__name = name

        self._employment.link(employer, self)

    @property
    def number(self) -> 'int':
        return self.__number

    @property
    def name(self) -> 'str':
        return self.__name

    @property
    def employer(self) -> 'Organization':
        return self._employment.employer_of(self)

    @property
    def employer_name(self) -> 'str':
        return self.employer.name

    @property
    def self_employee(self) -> 'Employee':
        return self

    @property
    def alias(self) -> 'str':
        return self.name

    def __str__(self) -> 'str':
        return "%s(number=%s, name=%s, employer_name=%s, alias=%s)" % (
            type(self).__name__,
            self.number,
            self.name,
            self.employer_name,
            self.alias
        )

File: cml-compiler/cml-generator/src/test/resources/templates/lang/java/expressions/employee.java
class Employee
{
    private static Employment employment;

    private final int number;
    private final String name;

    public Employee(int number, String name, Organization employer)
    {
        this.number = number;
        this.name = name;

        employment.link(employer, this);
    }

    public int getNumber()
    {
        return this.number;
    }

    public String getName()
    {
        return this.name;
    }

    public Organization getEmployer()
    {
        return employment.employerOf(this).get();
    }

    public String getEmployerName()
    {
        return getEmployer()
                   .getName();
    }

    public Employee getSelfEmployee()
    {
        return this;
    }

    public String getAlias()
    {
        return getName();
    }

    public String toString()
    {
        return new StringBuilder(Employee.class.getSimpleName())
                   .append('(')
                   .append("number=").append(String.format("\"%s\"", getNumber())).append(", ")
                   .append("name=").append(String.format("\"%s\"", getName())).append(", ")
                   .append("employerName=").append(String.format("\"%s\"", getEmployerName())).append(", ")
                   .append("alias=").append(String.format("\"%s\"", getAlias()))
                   .append(')')
                   .toString();
    }

    static void setEmployment(Employment association)
    {
        employment = association;
    }

    static
    {
        Employment.init(Employee.class);
    }
}